include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -437,6 +464,20 @@ __pattern_adjacent_find_ranges(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy
 
     return std::ranges::borrowed_iterator_t<_R>(std::ranges::begin(__r) + __idx);
 }
+
+template <typename _BackendTag, typename _ExecutionPolicy, typename _R, typename _Proj, typename _Comp>
+bool
+__pattern_is_sorted(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _R&& __r, _Comp __comp, _Proj __proj)
+{
+    auto __pred_2 = [__comp, __proj](auto&& __val1, auto&& __val2) { return std::invoke(__comp,
+        std::invoke(__proj, std::forward<decltype(__val1)>(__val1)),
+        std::invoke(__proj, std::forward<decltype(__val2)>(__val2)));};
```

Comment:
What about device copyability for this lambda?  
Unless we get changes in sycl 2020 spec about device copyable, we need this (and similar lambdas) to be structures with device_copyable specializations.  

If we don't do this, then any `__comp` or `__proj` which contains a `std::tuple` or other similar which is device_copyable but not trivially copyable will ruin the device copy-ability of this lambda, and therefore `__reorder_pred`.

Reply:
I forgot that we had an agreement to deal with these lambdas in a case-by-case manner for now as issues are reported instead of aggressively removing lambdas used like this from oneDPL.  Hopefully we can get traction for an update to automatically take care of these with a better general device copyable definition in SYCL itself.  
You are of course welcome to address this, but it is optional.  Resolving this comment.